src/core_plugins/timelion/public/directives/timelion_expression_input.js

```diff
@@ -109,6 +111,19 @@ app.directive('timelionExpressionInput', function ($document, $http, $interval,
             setCaretOffset(newCaretOffset);
             break;
           }
+          case SUGGESTION_TYPE.ARGUMENT_VALUE: {
+            const argumentName = `${scope.suggestions.list[suggestionIndex].name}`;
+            const { min, max } = suggestibleFunctionLocation;
+
+            // Update the expression with the function.
+            const updatedExpression = insertAtLocation(argumentName, scope.sheet, min, max);
+            scope.sheet = updatedExpression;
+
+            // Position the caret after the '='
+            const newCaretOffset = min + argumentName.length;
+            setCaretOffset(newCaretOffset);
+            break;
+          }
```

Comment:
There's a lot of repetition here. I think this becomes a little more maintainable if we extract the repetitive parts. I haven't tested this btw.

```javascript
        const { min, max } = suggestibleFunctionLocation;
        let insertedValue;
        let insertPositionMinOffset = 0;

        switch (scope.suggestions.type) {
          case SUGGESTION_TYPE.FUNCTIONS: {
            // Position the caret inside of the function parentheses.
            insertedValue = `${scope.suggestions.list[suggestionIndex].name}()`;

            // Update the expression with the function.
            // min advanced one to not replace function '.'
            insertPositionMinOffset = 1;
            break;
          }
          case SUGGESTION_TYPE.ARGUMENTS: {
            // Position the caret after the '='
            insertedValue = `${scope.suggestions.list[suggestionIndex].name}=`;
            break;
          }
          case SUGGESTION_TYPE.ARGUMENT_VALUE: {
            // Position the caret after the '='
            insertedValue = `${scope.suggestions.list[suggestionIndex].name}`;
            break;
          }
        }

        const updatedExpression = insertAtLocation(insertedValue, scope.sheet, min + insertPositionMinOffset, max);
        scope.sheet = updatedExpression;

        const newCaretOffset = min + insertedValue.length;
        setCaretOffset(newCaretOffset);
```

Reply:
I had meant to go back and out the duplicated code to clean this switch statement but just forgot about it. Thanks for cleaning it up. Your code block worked great